Allow space characters in material library name

dback2 opened the pull request:
This is to fix the parsing of the material library name token for certain Poly objects that have space in the material library filename.

Test Plan:
https://highfidelity.testrail.net/index.php?/cases/view/6776

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5511/HighFidelity-Beta-PR11951-8b38c03a5624f3fab6981214964628e227021667.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5511/HighFidelity-Beta-PR11951-8b38c03a5624f3fab6981214964628e227021667.dmg

cain-kilgore commented:
Testing.

cain-kilgore commented:
# Test Passed
https://highfidelity.testrail.net/index.php?/runs/view/1438&group_by=cases:section_id&group_order=asc
![image](https://user-images.githubusercontent.com/898687/33790847-e224a2f6-dc7b-11e7-9bcc-a9f471fa51f0.png)

elisa-lj11 commented:
@dback2 @cain-kilgore it looks like the Moon Buggy rendered correctly (this is the object where the material filename has a space in it: "Moon Buggy.mtl"), but the other files are not rendering properly, likely because their .mtl filename does not have a space in it.

Nex-Pro commented:
@dback2 Good fix and much needed.

Nex-Pro commented:
@elisa-lj11 This fix is to make sure that material library names can have spaces or no spaces isn't it ? 
I think it worked before where there were no spaces. Previously the moon buggy would always appear as grey only.

sethalves commented:
Could we see this same screenshot with master?

dback2 commented:
I think Elisa is referring to the Spinner which should not appear as grey only at least it didn't in my tests and has no space in its material library filename.

Nex-Pro commented:
@dback2 @sethalves @elisa-lj11  due to the large number of duplicate and similar but not identical files each with the same name on blocks  it would be better to have actual sample files. This may simply be a case of same name different file.

Nex-Pro commented:
Or maybe that was Clara.io

conklin94122 commented:
Does this PR need a milestone?

dback2 commented:
Not sure why the image guides that I had added to the test plan weren't working when I just checked, but I've re-added them and added an additional one to show the exact Spinner object and have verified the Spinner object should show correctly in world.

Ready for retest.

YuppyPlays commented:
Testing.

YuppyPlays commented:
Testing.

YuppyPlays commented:
Testing complete:

**Pass**

TestRail Results: https://highfidelity.testrail.net/index.php?/tests/view/24340&group_by=cases:section_id&group_order=asc&group_id=1459
